db/fetcher/fetcher.go

```diff
@@ -332,6 +332,10 @@ func (df *DocumentFetcher) nextKey(ctx context.Context, seekNext bool) (spanDone
 
 	df.kvEnd = spanDone
 	if df.kvEnd {
+		err = df.kvResultsIter.Close()
```

Comment:
Question: where was this close happening before? Do we have to worry about double closing now?

Reply:
Double closing is a non-issue, the docs say you can close as often as you like.  Before it was happening in fetcher.Close, and fetcher.startNextSpan.  It is still done in those places. In some situations (like the uncommented out test) Close will now be called multiple times.